mlir/lib/Dialect/LLVMIR/IR/LLVMMemorySlot.cpp

```diff
@@ -384,16 +468,17 @@ bool LLVM::GEPOp::canRewire(const DestructurableMemorySlot &slot,
   // dynamic indices can never be properly rewired.
   if (!getDynamicIndices().empty())
     return false;
+  //// TODO: This is not necessary, I think.
+  // if (slot.elemType != getElemType())
+  //   return false;
```

Comment:
This seem to be a debug left over and the condition seems to be checked above?

Maybe not in this PR, but should we use the data layout here as well instead of checking the element type. I.e. we could compute the offset of the GEP and then check if the offset points to the beginning of a subelement (of the slot and maybe also of a nested slot).

Reply:
Whups, this is indeed a rebase/stash artefact.
I have some changes in the pipeline that will cover this case.